tools/scripts/build/ensure-env.js

```diff
@@ -45,6 +45,12 @@ if (FREECODECAMP_NODE_ENV === 'production') {
       throw Error(`Env. variable ${key} is missing, build cannot continue`);
     }
   }
+
+  if (env['environment'] !== 'production')
+    throw Error("Production environment should be 'production' ");
+
+  if (env['showUpcomingChanges'])
```

Comment:
@ojeytonwilliams I did not dig to deep into the code, but is `showUpComdingChanges` a string (from an environment variable) or is it actually a Boolean here?  If the former, then this check would only be `false` if it is an empty string.

Reply:
It's a boolean by this point,  `showUpcomingChanges: showUpcomingChanges === 'true'`, the main thing is that it should never be truthy in production.